pallets/avn/src/tests/session_handler_tests.rs

```diff
@@ -1,167 +1,272 @@
+    add_collator(&new_candidate_id, auth_id);
+}
 
-    advance_session_and_force_new_validators();
+fn remove_collator_candidate(id: AccountId, validators_count: u32) {
+    let new_candidate_id = id;
+    remove_collator(&new_candidate_id, validators_count);
 }
 
-fn advance_session_no_validators_change() {
-    VALIDATORS.with(|v| {
-        let mut v = v.borrow_mut();
-        *v = None;
-        Some(v.clone())
-    });
+mod chain_started_with_initial_colators {
+    use super::*;
 
-    advance_session_and_force_new_validators();
-}
+    fn setup_initial_collators() -> sp_io::TestExternalities {
+        let initial_validators_staking: Vec<(sp_core::sr25519::Public, u128)> = vec![
+            (TestAccount::derive_account_id(1), 100),
+            (TestAccount::derive_account_id(2), 100),
+            (TestAccount::derive_account_id(3), 100),
+            ];
 
-fn advance_session_and_force_new_validators() {
-    // need to do it twice for the change to take effect
-    advance_session();
-    advance_session();
-}
+        let initial_validators_session:Vec<u64> = vec![1,2,3];
 
-// TODO [TYPE: test refactoring][PRI: LOW]:  update this function to work with the mock builder
-// pattern Currently, a straightforward replacement of the test setup leads to an error on the
-// assert_eq!
-fn advance_session() {
-    let now = System::block_number().max(1);
-    System::set_block_number(now + 1);
-    Session::rotate_session();
-    assert_eq!(Session::current_index(), (now / Period::get()) as u32);
-}
+        let initial_account_balances: Vec<(AccountId, Balance)> = vec![
+            (TestAccount::derive_account_id(1), 10000),
+            (TestAccount::derive_account_id(2), 10000),
+            (TestAccount::derive_account_id(3), 10000),
+            (TestAccount::derive_account_id(4), 10000),
+            (TestAccount::derive_account_id(5), 10000),
+            (TestAccount::derive_account_id(6), 10000),
+            (TestAccount::derive_account_id(7), 10000),
+            (TestAccount::derive_account_id(8), 10000),
+            (TestAccount::derive_account_id(9), 10000),
+            (TestAccount::derive_account_id(10), 10000),
+            ];
 
-#[test]
-//* good case: keys have been imported in the ethereum-events pallet
-fn keys_populated_correctly_on_genesis() {
-    let mut ext = ExtBuilder::build_default().with_validators().as_externality();
-    ext.execute_with(|| {
-        assert!(
-            AVN::validators() ==
-                vec![
-                    Validator { account_id: 1, key: UintAuthorityId(1) },
-                    Validator { account_id: 2, key: UintAuthorityId(2) },
-                    Validator { account_id: 3, key: UintAuthorityId(3) }
-                ]
-        );
-    });
-}
+        let mut ext = ExtBuilder::build_default().with_balances(initial_account_balances.clone()).with_validators(initial_validators_session).with_staking(initial_validators_staking).as_externality();
```

Comment:
Probably would look better, and read better, with each operation in a newline.
```suggestion
        let mut ext = ExtBuilder::build_default()
        
    .with_balances(initial_account_balances.clone())
    .with_validators(initial_validators_session)
    .with_staking(initial_validators_staking)
    .as_externality();
```